aries_vcx/tests/test_pool.rs

```diff
@@ -68,22 +69,34 @@ mod integration_tests {
         .await;
     }
 
-    // TODO - future - bring back after all endorser methods added to baseledger
-    // #[tokio::test]
-    // async fn test_endorse_transaction() {
-    //     SetupProfile::run_indy(|setup| async move {
-    //         let ledger = Arc::clone(&setup.profile).inject_ledger();
-    //         let (author_did, _) = add_new_did(&setup.profile, &setup.institution_did, None).await.unwrap();
-    //         let (endorser_did, _) = add_new_did(&setup.profile, &setup.institution_did, Some("ENDORSER")).await.unwrap();
-
-    //         let schema_request = ledger.build_schema_request(&author_did, SCHEMA_DATA).await.unwrap();
-    //         let schema_request = append_request_endorser(&schema_request, &endorser_did).await.unwrap();
-    //         let schema_request = multisign_request(setup.wallet_handle, &author_did, &schema_request)
-    //             .await
-    //             .unwrap();
-    //         ledger.endorse_transaction(&endorser_did, &schema_request).await.unwrap();
-    //     }).await;
-    // }
+    #[tokio::test]
+    #[ignore]
+    async fn test_endorse_transaction() {
+        SetupProfile::run_indy(|setup| async move {
+            let ledger = Arc::clone(&setup.profile).inject_ledger();
+            let (author_did, _) = add_new_did(&setup.profile, &setup.institution_did, None).await.unwrap();
+            let (endorser_did, _) = add_new_did(&setup.profile, &setup.institution_did, Some("ENDORSER"))
+                .await
+                .unwrap();
+
+            let schema_request = ledger.build_schema_request(&author_did, SCHEMA_DATA).await.unwrap();
+            let schema_request = ledger
+                .set_endorser(&author_did, &schema_request, &endorser_did)
```

Comment:
I know this is just restored code, and my comment could be considered out of scope, but I'll leave it here:

It's kind of awkward workflow as you need to set endorser on transaction ahead of time, then try to call `endorse_transaction` which does verification that `set_endorser` has been applied. All while doing de/serialization from/to strings.

Would be good to either address it while we are at it, or create issue for this.

Maybe I am missing some nuance, but it seems to getting rid of `set_endorser` me just having `endorse_transaction` with additional `endorser_did` parameter would make things a lot simpler

Reply:
I agree. I don't know why it is split like this.

Reply:
Do we agree addressing this is outside of this PR's scope?

Reply:
Yeah we can do that. Would you create issue for it?